src/core/probably.scala

```diff
@@ -112,23 +112,31 @@ class Runner(specifiedTests: Set[TestId] = Set()) extends Dynamic {
 
   abstract class Test(val name: String, map: => Map[String, String]) {
     type Type
-    
+
     def id: TestId = TestId(Runner.shortDigest(name))
     def action(): Type
-    
-    def assert(predicate: Type => Boolean): Unit =
-      try if(runTest(id)) check(predicate) catch { case NonFatal(e) => () }
-    
-    def check(predicate: Type => Boolean): Type = {
+
+    def assert(predicate: (Type => Boolean)*): Unit =
+      try if(runTest(id)) check(predicate : _*) catch { case NonFatal(_) => () }
+
+    def check(predicates: (Type => Boolean)*): Type = {
+      def handler(index: Int): PartialFunction[Throwable, Datapoint] = { case e: Exception => ThrowsInCheck(e, map, index) }
+      def makeDatapoint(predicates: Seq[Type => Boolean], count: Int, datapoint: Datapoint, value: Type): Datapoint =
+        try {
+          if (predicates.isEmpty) datapoint
+          else
+            if (predicates.head(value)) makeDatapoint(predicates.tail, count + 1, Pass, value)
+            else Fail(map, count)
+        } catch handler(count)
+
+
       val t0 = System.currentTimeMillis()
       val value = Try(action())
       val time = System.currentTimeMillis() - t0
 
       value match {
         case Success(value) =>
-          def handler: PartialFunction[Throwable, Datapoint] = { case e: Exception => ThrowsInCheck(e, map) }
-          val datapoint: Datapoint = try(if(predicate(value)) Pass else Fail(map)) catch handler
-          record(this, time, datapoint)
+          record(this, time, makeDatapoint(predicates, 0, Pass, value))
```

Comment:
```suggestion
```

Reply:
there is another instruction in the next line, should we leave it that way?
```scala
case Success(value) =>
          record(this, time, makeDatapoint(predicates, 0, Pass, value))
          value
```